File: src/main/java/org/pl/parser/ast/IfNode.java
package org.pl.parser.ast;

import java.util.List;

public class IfNode implements INode {
    public final List<IfCase> cases;
    public final INode elseCase;

    public IfNode(List<IfCase> cases) {
        this(cases, null);
    }

    public IfNode(List<IfCase> cases, INode elseCase) {
        this.cases = cases;
        this.elseCase = elseCase;
    }

    @Override
    public String toString() {
        return "IfNode{" +
                "cases=" + cases +
                ", elseCase=" + elseCase +
                '}';
    }
}

File: src/main/kotlin/org/nyxlang/parser/ast/IfNode.kt
package org.nyxlang.parser.ast

/**
 * A conditional if node that contains if cases with conditions and bodies
 * and an alternative else case.
 */
class IfNode(val cases: List<IfCase>, val elseCase: INode?) : INode {
    override fun toString() = "IfNode(cases=$cases, else=$elseCase)"
}